Boundary Docs: adds config KMS usage docs (#3164)
hashicorp/web-unified-docs: content/boundary/v0.19.x/content/docs/secure/encryption/data-encryption.mdx

encrypted values found at startup time.

You can configure `config` KMS keys for self-managed Enterprise or Community edition deployments.
Unlike the other keys described on this page, the `config` key does not protect
data that Boundary stores. It protects the configuration file itself, before
Boundary reads it. As a result, the `config` key is the only key that both an
operator and the Boundary server need to access: the operator uses it to encrypt
values, and Boundary uses it to decrypt them at startup.

### When to use a config KMS key

Consider a `config` KMS key when your configuration file contains values you
cannot store in plaintext, and the file itself must live somewhere you do not
fully control. Common cases include:

- Committing configuration files to version control.
- Distributing configuration through a configuration management system.
- Building configuration into an artifact in a CI/CD pipeline.

Encrypting configuration values is not a substitute for restricting access to

> the configuration file. It narrows the exposure if the file is read by someone
> who cannot reach the KMS.

Reviewer: ```suggestion
the configuration file. It narrows exposure by preventing someone
who cannot reach the KMS from reading the file.
```

Reviewer: Passive voice
